dialogue-core/src/test/java/com/palantir/dialogue/core/ChannelsTest.java

```diff
@@ -138,4 +140,20 @@ public void bad_channel_throwing_an_error_still_returns_a_future() {
         // only when we access things do we allow exceptions
         assertThatThrownBy(() -> Futures.getUnchecked(future)).hasCauseInstanceOf(NoClassDefFoundError.class);
     }
+
+    @Test
+    @TestTracing
+    @SuppressWarnings("FutureReturnValueIgnored")
+    public void traces_on_retries() {
+        when(response.code()).thenReturn(429);
+        channel.execute(endpoint, request);
```

Comment:
Feels a little less brittle instead of `@SuppressWarnings("FutureReturnValueIgnored")`
```suggestion
        channel.execute(endpoint, request).get();
```